jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFComponentProcessor.java

```diff
@@ -67,13 +73,149 @@
 
     private static final String WEB_INF_FACES_CONFIG = "WEB-INF/faces-config.xml";
 
-    private static final String MANAGED_BEAN = "managed-bean";
-    private static final String MANAGED_BEAN_CLASS = "managed-bean-class";
-    private static final String LIFECYCLE = "lifecycle";
-    private static final String PHASE_LISTENER = "phase-listener";
-
     private static final String CONFIG_FILES = "javax.faces.CONFIG_FILES";
 
+    /**
+     * JSF tags that should be checked in the configuration file. All the
+     * tags that are needed for injection are taken into account.
+     * If more artifacts should be included just add it to the enum and to
+     * the <em>facesConfigElement</em> tree in order to be parsed.
+     */
+    private enum JsfTag {
+      FACES_CONFIG,
+        FACTORY,
+          APPLICATION_FACTORY,
```

Comment:
Can all the whitespace issues in this PR be cleaned up please? Thanks.

Reply:
OK, I'll re-format the enum when I know what to do with the message to do both things ate the same time. This was done on purpose, to see the XML structure. :-)

Reply:
Fixed!